### PR TITLE
feat: profile and referral feat (cont)

### PR DESCRIPTION
**Pull request by FME849**

### CODE REVIEW

File: api/user/index.ts [hunk 1]
```diff
@@ -1,27 +1,66 @@
...
+		address: dynamoUser.address || '',
+		jwt: dynamoUser.jwt || '',
+		name: dynamoUser.name || '',
+		email: dynamoUser.email || '',
+		githubId: dynamoUser.githubId || '',
+		githubUrl: dynamoUser.githubUrl || '',
+		avatarUrl: dynamoUser.avatar || '',
+		mineral: dynamoUser.mineral || 0.0,
+		referralCode: generateRandomCode(REFERRAL_CODE_LENGTH),
+	});
```
Comment (FME849) [inline api/user/index.ts:32]: @tanlethanh I cloned the data from Dynamo to Mongo here, please have a look and feedback

Comment (tanlethanh) [reply api/user/index.ts:32]: I think we only keep the relavent fields, could remove `github`fields, `mineral`

Comment (tanlethanh) [reply api/user/index.ts:32]: also, these code should be in `graphql context`, all resolvers shouldn't know about dynamodb

File: api/user/index.ts [hunk 2]
```diff
@@ -1,27 +1,66 @@
...
+		avatarUrl: user.avatarUrl,
+		mineral: user.mineral,
+		referralCode: user.referralCode,
 	};
+};
+
+export const refereeUser: ReferralHistoryResolvers['refereeUser'] = async ({
+	refereeId,
+}) => {
+	const { Item: dynamoProfile } = await getItem(`profile#${refereeId}`);
```
Comment (tanlethanh) [inline api/user/index.ts:56]: no dynamodb here

File: api/models/user.ts [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
 
+export type IUser = {
+	id: string;
+	linkedId: string;
+	address: string;
+	jwt: string;
+	name: string;
+	email: string;
+	githubId: string;
+	githubUrl: string;
```
Comment (tanlethanh) [inline api/models/user.ts:13]: should remove these fields

File: api/models/user.ts [hunk 4]
```diff
@@ -1,8 +1,21 @@
 import { model } from 'mongoose';
-import { generateRandomCode } from 'utils/referral';
 
 import { createSchema } from './utils';
 
+export type IUser = {
+	id: string;
+	linkedId: string;
```
Comment (tanlethanh) [inline api/models/user.ts:7]: what is linkedId?

File: api/models/user.ts [hunk 5]
```diff
@@ -1,8 +1,21 @@
...
+	address: string;
+	jwt: string;
+	name: string;
+	email: string;
+	githubId: string;
+	githubUrl: string;
+	avatarUrl: string;
+	mineral: number;
```
Comment (tanlethanh) [inline api/models/user.ts:15]: should remove this field

**Pull request merged**